include/deal.II/distributed/cell_weights.h

```diff
@@ -38,80 +38,178 @@ namespace parallel
    * a hp::DoFHandler. One can choose from predefined weighting
-   * algorithms provided by this class or provide a custom one. The
-   * chosen weighting function will be connected to the corresponding
-   * signal of the linked parallel::TriangulationBase via callback.
+   * algorithms provided by this class or provide a custom one.
    *
-   * An object of this class needs to exist for every DoFHandler associated
-   * with the Triangulation we work on to achieve satisfying work balancing
-   * results.
-   *
-   * A small code snippet follows explaining how to achieve each cell
-   * being weighted by its current number of degrees of freedom.
+   * This class offers two different ways of connecting the chosen weighting
+   * function to the corresponding signal of the linked
+   * parallel::TriangulationBase. The recommended way involves creating an
+   * object of this class which will automatically take care of registering the
+   * weighting function upon creation and de-registering it once destroyed. An
+   * object of this class needs to exist for every DoFHandler associated with
+   * the Triangulation we work on to achieve satisfying work balancing results.
+   * The following code snippet demonstrates how to achieve each cell being
+   * weighted by its current number of degrees of freedom. The connected
+   * weighting function may be changed anytime using the CellWeights::reinit()
+   * function.
    * @code
-   * parallel::CellWeights<dim, spacedim> cell_weights(hp_dof_handler);
-   * cell_weights.register_ndofs_weighting();
+   * parallel::CellWeights<dim, spacedim> cell_weights(
+   *   hp_dof_handler, parallel::CellWeights<dim, spacedim>::ndofs_weighting());
    * @endcode
-   * The weighting function can be changed anytime. Even more ambitious
-   * approaches are possible by submitting customized functions, e.g.
+   *
+   * On the other hand, you are also able to take care of handling the signal
+   * connection manually by using the static member function of this class. In
+   * this case, an analogous code example looks as follows.
    * @code
-   * cell_weights.register_custom_weighting(
-   *   [](const FiniteElement<dim, spacedim> &active_fe,
-   *      const typename hp::DoFHandler<dim, spacedim>::cell_iterator &)
-   *   -> unsigned int {
-   *   return 1000 * std::pow(active_fe.dofs_per_cell, 1.6);
-   * });
+   * boost::signals2::connection listener =
+   *   dof_handler.get_triangulation().signals.cell_weight.connect(
+   *     parallel::CellWeights<dim, spacedim>::make_weighting_callback(
+   *       dof_handler,
+   *       parallel::CellWeights<dim, spacedim>::ndofs_weighting());
    * @endcode
-   * The returned value has to be an unsigned integer and is thus limited by
-   * some large number. It is interpreted as the additional computational load
-   * of each cell. See Triangulation::Signals::cell_weight for a discussion on
-   * this topic.
```

Comment:
Would it be useful to preserve these four lines somehow, or at least link to a place where it is explained how the weighting works?

Reply:
Are you okay with moving the explanation to the `WeightingFunction` alias (as you suggested below) and leaving a `@note` referring to `Triangulation::Signals::cell_weight` in the general class documentation?